src/python/pants/backend/debian/rules.py

```diff
@@ -36,24 +37,38 @@ async def package_debian_package(field_set: DebianPackageFieldSet) -> BuiltPacka
     dpkg_deb_path = await Get(
         BinaryPaths,
         BinaryPathRequest(
-            binary_name="touch",
+            binary_name="dpkg-deb",
             search_path=["/usr/bin"],
         ),
     )
     if not dpkg_deb_path.first_path:
-        raise EnvironmentError("Could not find the `touch` program on search paths ")
+        raise EnvironmentError("Could not find the `dpkg-deb` program on search paths ")
 
-    output_filename = field_set.output_path.value_or_default(field_set.address, file_ending="deb")
+    output_filename = field_set.output_path.value_or_default(
+        field_set.address,
+        file_ending="deb",
+    )
+
+    # TODO: need to create a digest from the directory from the "sources" field of
+    #  "debian_package" target, i.e. field_set.control.filespec['includes'][0]
+    package_metadata_dir_digest = await Get(Digest, CreateDigest("?"))
 
-    # TODO(alexey): add Debian packaging logic
     result = await Get(
         ProcessResult,
         Process(
             argv=(
-                "touch",
-                output_filename,
+                "/usr/bin/dpkg-deb",
+                "--build",
+                # this is assumed to be a directory from the "sources" field of
+                # "debian_package" target
+                field_set.control.filespec["includes"][0],
             ),
             description="Create a Debian package from the produced packages.",
+            input_digest=package_metadata_dir_digest,
+            # TODO: don't need the working_dir since the "sources" will become available?
+            #  and the output .deb file will become available as a BuiltPackage because
+            #  it has been specified in the "output_files"?
+            working_dir="?",
```

Comment:
@Eric-Arellano will the .deb file be created in the root of the repository? Perhaps we should set up the output directory to be `dist/` as it happens with PEX files.

Reply:
This proces will run in a temporary directory (chroot). You shouldn't need to set `working_dir`. Instead, you only need `output_files` to be the right thing relative to that tmpdir's root. So if you're running `dpkg build -o result.deb`, then you'd set `output_files=('result.deb',)`. 

You'll now have the `process.output_digest` as a handle to that single file, and that file can be moved all around and manipulated like adding a prefix to it. The `package` goal will use the digest to write it to `dist/` for you.

To better understand `Digest`s, it can help to "enrich" them to be `Snapshot`s.https://www.pantsbuild.org/docs/rules-api-file-system#core-abstractions-digest-and-snapshot